devolutions-gateway/src/streaming/mod.rs

```diff
@@ -0,0 +1,69 @@
+use anyhow::Context;
+use axum::{body::Body, response::Response};
+use streamer::{webm_stream, ReOpenableFile};
+use uuid::Uuid;
+
+use crate::{recording::OnGoingRecordingState, token::RecordingFileType, ws::websocket_compat};
+
+struct ShutdownSignal(devolutions_gateway_task::ShutdownSignal);
+
+impl streamer::Signal for ShutdownSignal {
+    fn wait(&mut self) -> impl std::future::Future<Output = ()> + Send {
+        self.0.wait()
+    }
+}
+
+pub(crate) async fn stream_file(
+    path: camino::Utf8PathBuf,
+    ws: axum::extract::WebSocketUpgrade,
+    shutdown_signal: devolutions_gateway_task::ShutdownSignal,
+    recordings: crate::recording::RecordingMessageSender,
+    recording_id: Uuid,
+) -> anyhow::Result<Response<Body>> {
+    // 1.identify the file type
+    if path.extension() != Some(RecordingFileType::WebM.extension()) {
+        return Err(anyhow::anyhow!("Invalid file type"));
```

Comment:
This is a nitpick that applies to many place in your PR: https://github.com/Devolutions/IronRDP/blob/fe0d9e9773d96ddb7b51c0c4e0d42821d0da569b/STYLE.md#formatting-of-error-messages
If you have time to address it do it, otherwise can wait

Reply:
I checked with ripgrep, should be all fixed